### PR TITLE
Heroku Deploy

### PR DESCRIPTION
**Pull request by loicteixeira**
Use and configure `rails_12factor`, `figaro` and `puma` for Heroku deployment.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -13,6 +13,9 @@ gem 'sdoc', '~> 0.4.0', group: :doc
```
Comment (eliotsykes) [inline Gemfile]: The `pg` version string looks older than necessary - would have expected 0.18

**Comment by eliotsykes** [comment]
All good. No need to wait for feedback to merge PR if it holds up your progress.

File: Gemfile (hunk 1 above)
Comment (loicteixeira) [inline Gemfile]: That is what rails 4.2.5.1 generates with `rails new my_app -d postgresql`.
Do we have a good reason to think they picked this version on purpose?

**Comment by loicteixeira** [comment]
Closing to recreate the PR with the target set to the `master` branch.

**Pull request closed without merging**